File: cosmowcl/src/app/shared/services/tipi-notifiche.service.ts
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { TipoNotifica } from '../models/api/cosmonotifications/tipoNotifica';
import { ApiUrls } from '../utilities/apiurls';

@Injectable({
  providedIn: 'root'
})
export class TipiNotificheService {

  constructor(private http: HttpClient) { }

  getTipiNotifiche(): Observable<TipoNotifica[]>{
    return this.http.get<TipoNotifica[]>(ApiUrls.TIPI_NOTIFICHE);
  }
}

File: cosmonotifications/src/java/it/csi/cosmo/cosmonotifications/business/service/TipiNotificheService.java
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

package it.csi.cosmo.cosmonotifications.business.service;

import java.util.List;
import it.csi.cosmo.cosmonotifications.dto.rest.TipoNotifica;

/**
 *
 */

public interface TipiNotificheService {

  List<TipoNotifica> getTipiNotifiche();

}
